src/main/java/school/lemon/changerequest/java/introduction/hw2/Task4.java

```diff
@@ -2,6 +2,26 @@
 
 public class Task4 {
     public static String charAt(String s, int i) {
-        return "";
+
+
+        if (i < 0 && Math.abs(i) <= s.length()) {
+            StringBuilder s1 = new StringBuilder(s);
+            s1.reverse();
+            i = Math.abs(i) - 1;
+            return String.valueOf(s1.charAt(i));
+        }
+
+
+        if (i < 0 && Math.abs(i) > s.length()) {
+            StringBuilder s1 = new StringBuilder(s);
+            s1.reverse();
+            i = s.length() - (Math.abs(i) - 1);
+            return String.valueOf(s1.charAt(i));
+        }
+
+        if (i > 0 && i >= s.length()) {
+            i = i - s.length();
+        }
+        return String.valueOf(s.charAt(i));
```

Comment:
Your implementation pass all tests, but you've used lots of unnecessary operations and it'll be quite slow (due to additional usage of `Math` and `StringBuilder`.
Try to solve this task from Math point of view - you just need to calculate char index properly and this could be done using `%` operator.